operators/tokenizer/bpe_utils.hpp

```diff
@@ -198,7 +198,8 @@ class TokenWithRegularExp {
     std::wsmatch match;
 
     if (std::regex_search(wstr, match, pattern)) {
-        std::u32string_view token = std::u32string_view(W2Ustring(match.str()).data(), match.str().size());
+        std::u32string temp = W2Ustring(match.str());
+        std::u32string_view token = std::u32string_view(temp.data(), match.str().size());
```

Comment:
But it is still a temporary object. Why do you return it out at line 204?

Reply:
so `W2Ustring(match.str())` creates a temporary std::u32string object right, and the std::u32string_view token was created from its data, so once the temporary std::u32string is destroyed, the std::u32string_view becomes invalid.

hence i add another step and create the temp variable to ensure that the std::u32string object outlives the std::u32string_view

Reply:
@sayanshaw24, @snnn is right, there is another issue of returning a view object refereeing to a temporary object.
![image](https://github.com/user-attachments/assets/8be1002b-5295-44c1-b246-4bd0853e8b51). You didn't fix the issue completely.